forklift/LocatorPallet.py

```diff
@@ -0,0 +1,370 @@
+
+                    return False
+
+                self.success = (True, f'{", ".join(dirty_locators)} were deployed')
+
+                return True
+
+    def _get_dirty_locators(self):
+        lookup = {
+            'addresspoints': 'AddressPoints_AddressSystem',
+            'atlnamesaddrpnts': 'AddressPoints_AddressSystem',
+            'atlnamesroads': 'Roads_AddressSystem_STREET',
+            'geocoderoads': 'Roads_AddressSystem_STREET',
+        }
+
+        return set([lookup[crate.source_name.lower()] for crate in self.get_crates() if crate.was_updated()])
+
+    def copy_locator_to(self, file_path, locator, to_folder):
+        location = Path(file_path) / locator
+        self.log.debug('copying %s to %s', location, to_folder)
+        #: iterator glob for .loc .loz
+        for filename in iglob(str(location) + '.lo*'):
+            _, locator_with_extension = split(filename)
+
+            to_folder.mkdir(parents=True, exist_ok=True)
+
+            output = to_folder / locator_with_extension
+
+            copyfile(filename, output)
+
+    def create_locators(self):
+        #: address points
+        primary_fields = [
+            "'PointAddress.FEATURE_ID AddressPoints.OBJECTID'",
+            "'PointAddress.ADDRESS_JOIN_ID AddressPoints.UTAddPtID'",
+            "'PointAddress.HOUSE_NUMBER AddressPoints.AddNum'",
+            "'PointAddress.BUILDING_NAME AddressPoints.LandmarkName'",
+            "'PointAddress.STREET_NAME_JOIN_ID AddressPoints.UTAddPtID'",
+            "'PointAddress.STREET_PREFIX_DIR AddressPoints.PrefixDir'",
+            "'PointAddress.STREET_NAME AddressPoints.StreetName'",
+            "'PointAddress.STREET_SUFFIX_TYPE AddressPoints.StreetType'",
+            "'PointAddress.STREET_SUFFIX_DIR AddressPoints.SuffixDir'",
+            "'PointAddress.FULL_STREET_NAME AddressPoints.FullAdd'",
+            "'PointAddress.SUB_ADDRESS_UNIT AddressPoints.UnitID'",
+            "'PointAddress.SUB_ADDRESS_UNIT_TYPE AddressPoints.UnitType'",
+            "'PointAddress.CITY AddressPoints.City'"
+        ];
+        alt_fields = [
+            "'AlternateHouseNumber.ADDRESS_JOIN_ID AtlNamesAddrPnts.UTAddPtID'",
+            "'AlternateHouseNumber.HOUSE_NUMBER AtlNamesAddrPnts.AddNum'",
+            "'AlternateStreetName.STREET_NAME_JOIN_ID AtlNamesAddrPnts.UTAddPtID'",
+            "'AlternateStreetName.STREET_PREFIX_DIR AtlNamesAddrPnts.PrefixDir'",
+            "'AlternateStreetName.STREET_NAME AtlNamesAddrPnts.StreetName'",
+            "'AlternateStreetName.STREET_SUFFIX_TYPE AtlNamesAddrPnts.StreetType'",
+            "'AlternateStreetName.STREET_SUFFIX_DIR AtlNamesAddrPnts.SuffixDir'"
+        ]
+
+        start_seconds = perf_counter()
+        process_seconds = perf_counter()
+        self.log.info('creating the %s locator', 'address point')
+        try:
+            output_location = str(Path(self.output_location) / 'AddressPoints_AddressSystem')
```

Comment:
**suggestion**: save properties as Path classes on the instance e.g. (self.output_location, self.locators) and call `str` when necessary to remove the calls to the path constructor?